cuda_bindings/cuda/bindings/_path_finder/README.md

```diff
@@ -0,0 +1,61 @@
+# `cuda.bindings.path_finder` Module
+
+## Public API (Work in Progress)
+
+Currently exposes two primary interfaces:
+
+```
+cuda.bindings.path_finder._SUPPORTED_LIBNAMES  # ('nvJitLink', 'nvrtc', 'nvvm')
+cuda.bindings.path_finder._load_nvidia_dynamic_library(libname: str) -> LoadedDL
+```
+
+**Note:**
+These APIs are prefixed with an underscore because they are considered
+experimental while undergoing active development, although already
+reasonably well-tested through CI pipelines.
+
+## Library Loading Search Priority
+
+The `load_nvidia_dynamic_library()` function implements a hierarchical search
+strategy for locating NVIDIA shared libraries:
```

Comment:
I think we should clarify that the current behavior is that the search is done on a per library basis and that there is no centralized mechanism that ensures all libraries are found from the same location / installation method. I.E. if someone has a `libcufft` wheel installed in a conda environment with `libcublas` installed with a system installation of the toolkit, we'll currently find `libcufft` from the wheel, `libcublas` from the conda environment, and the rest of our needed libraries from the system toolkit.

In the future we should probably not allow this behavior to occur, but for now we should probably clearly document that this is not behavior we intend to support in the future and is essentially undefined behavior as of now.

Reply:
FWIW in nvmath-python docs we simplified what Keith said into: "Don't mix-n-match pip, conda, and system (installations of CUDA)" 😄 https://docs.nvidia.com/cuda/nvmath-python/latest/installation.html#troubleshooting

Reply:
I added commit 6a60161d8ab6773295cd856d643880624b12d1ec.

> Don't mix-n-match

I expect that that's exactly what many people will have, knowingly or unknowingly. If we release system CTKs, conda CTKs, and CTK wheels, they will get mixed inevitably.

I believe the best we can do is to be accommodating, within reason.